examples/align/readme.md

```diff
@@ -17,4 +17,186 @@ The application should open a window and display a video stream from the camera.
+
+`try_get_depth_scale(..)` is a simple function that tries to find a depth sensor from the pipeline's device and retrieve its depth scale units.
+
+Heading to `main`:
+
+We first define some variables that will be used to show the window and render the images and GUI to the screen:
+
+```cpp
+// Create and initialize GUI related objects
+    window app(1280, 720, "CPP - Align Example"); // Simple window handling
+    ImGui_ImplGlfw_Init(app, false);      // ImGui library intializition 
+    rs2::colorizer c;                          // Helper to colorize depth images
+    texture renderer;                     // Helper for renderig images
+```
+
+Then, we create a context:
+
+```cpp
+// Create a context
+rs2::context ctx;
+```
+`rs2::context` encapsulates all of the devices and sensors, and provides some additional functionalities.
+In this example we will use the `context` to create an `rs2::align` object:
+
+```cpp
+    // Using the context to create a rs2::align object. 
+    // rs2::align allows you to perform aliment of depth frames to others
+    rs2::align align = ctx.create_align(RS2_STREAM_COLOR); 
+```
+
+Next, we define a `rs2::pipeline` to help us easily configure and start the camera.
+
+```cpp
+// Create a pipeline to easily configure and start the camera
+rs2::pipeline pipe;
+```
+
+We start the pipeline using an overloaded version of `start` which takes a callable object.
+
+```cpp
+// By passing align to Pipeline::start, frames will be passed to align's handler
+    pipe.start(align);
+```
+
+`rs2::align` contains an overload for `operator()` which takes a `rs2::frame`. This allows us to pass it as the argument for `pipeline::start`. The pipeline synchronizes its frames before publishing them, thus, by calling `start` with an `align` object, the frames it will provide will be synchronized and aligned to one another. We will soon see how to get these frames from the `align` object.
+
+At this point of the program the camera is started and frames are being aligned.
+
+Before actually using the frames, we try to get the depth scale units of the depth camera. Depth scale units represent the units in which depth pixels are represented. 
+
+These units are expressed as depth in meters corresponding to a depth value of 1. For example if we have a depth pixel with a value of 2 and the depth scale units are 0.5 then that pixel is `2 X 0.5 = 1` meter away from the camera.
+
+Now comes the interesting part of the application. We start our main loop, which breaks only when the window is closed:
+
+
+```cpp
+while (app) // Application still alive?
+{
+```
+
+Inside the loop, the first thing we do is block the program until the `align` object returns a `rs2::frameset`.
```

Comment:
Please elaborate what `rs2::frameset` provides, to remove the redundancy with the comment in the code below